pkg/apis/stork/v1alpha1/types.go

```diff
@@ -212,3 +217,193 @@ type MigrationList struct {
 
 	Items []Migration `json:"items"`
 }
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// StorageCluster represents a storage cluster
+type StorageCluster struct {
+	meta.TypeMeta   `json:",inline"`
+	meta.ObjectMeta `json:"metadata,omitempty"`
+	Spec            StorageClusterSpec   `json:"spec"`
+	Status          StorageClusterStatus `json:"status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// StorageClusterList is a list of StorageCluster
+type StorageClusterList struct {
+	meta.TypeMeta `json:",inline"`
+	meta.ListMeta `json:"metadata,omitempty"`
+	Items         []StorageCluster `json:"items"`
+}
+
+// StorageClusterSpec is the spec used to define a storage cluster
+type StorageClusterSpec struct {
+	// Image is docker image of the storage driver
+	Image string `json:"image"`
+	// Kvdb is the information of kvdb that storage driver uses
+	Kvdb KvdbSpec `json:"kvdb"`
+	// Network is the network information for storage driver
+	Network NetworkSpec `json:"network"`
+	// Storage details of storage used by the driver
+	Storage StorageSpec `json:"storage"`
+	// CloudStorage details of storage in cloud environment.
+	// Keep this only at cluster level for now until support to change
+	// cloud storage at level is added.
+	CloudStorage CloudStorageSpec `json:"cloudStorage"`
```

Comment:
Should CloudStorage be a part of the StorageSpec? StorageSpec should take care of whether storage is from local disks or cloud disks

Reply:
Had consciously separated them because there are different options in both them that don't make sense when combined with the other.
Like max_storage_nodes or max_storage_nodes_per_zone in storage spec,
and use_all, use_all_with_partitions in cloud spec.
Would be overall easier for people to understand what they are doing. They will specify only one of them.